### DISCUSSION TITLE
Performance optimization and custom commands

### DISCUSSION THREAD

**Discussion by yegorf** [eckig/graph-editor#208] [Q&A]
Hello, Steffen.
I have 4 questions:

1) Are there any general tricks to improve performance when working with large schemas? If there are no connections between the nodes, it works fine with a few hundred nodes. But if they are connected by a large number of connectors, the performance issue becomes much more significant. Currently, I have disabled mass node copy/paste and mass node moving via drag and drop (as these cause the biggest performance issues). Perhaps something else can be done (for example, resource cleanup)?
Also are there any ways to avoid an ANR (Application Not Responding) situation when the UI thread is overloaded?

2) I resize the schema with this code. 
`            val model = graphEditor.model

            model.contentWidth += paperPageSize.width

            graphEditor.setModel(model)

            graphEditorContainer.setGraphEditor(graphEditor)`

It works, but after a few resizes, the schema navigation starts lagging. I found that the problem is in the grid display. Should I clear it every time, or is there another way to change the model size?

3) What could be the reason why commands from AbstractOverrideableCommand are not added to the command stack and are not executed, but everything works when implementing the Command interface?

4) Can custom commands somehow be used for handling the state of JavaFX default controls (for example, adding/removing rows in a TableView)?

Thank you.

**Comment by eckig** [comment]
1. I am sorry to hear that. I have not developed an app that is so large, that the UI Thread gets overloaded. Can you, by any chance, create a [MCVE](https://stackoverflow.com/help/minimal-reproducible-example) to test this?
2. The grid display is indeed one problematic area, but I have not yet found a better way to optimize this.
3. How do you execute the commands?
4. In theory: Yes. But you would have to create custom listeners for these commands. You can create and use a subclass of `GraphEditorController` to do this. If you look at `initDefaultListeners()` that there are even some utilities for doing this.
